src/Swarrot/Broker/MessageProvider/MessageProviderInterface.php

```diff
@@ -13,6 +13,14 @@
      */
     public function get();
 
+    /**
+     * consume
+     *
+     * @param string $consumerTag Identify the consumer for the rabbitmq-server
+     * @param callable $callback A callback receiving a Swarrot\Broker\Message instance on consumption
+     */
+    public function consume($consumerTag, callable $callback);
```

Comment:
adding a new method in an existing interface will force making a new major version of the library, as it is a BC break for any implementor of the interface.

Thus, making it required in the interface would make the interface much harder to implement for any other backend. I suggest making part of a separate optional interface.
And then, handling the case of requesting the `consume` method in Consumer for a provider not implementing that interface could be done in multiple ways:
- throwing an exception saying it is unsupported
- silently switching to `get`
- wrapping it into an adapter which would implement a loop in PHP to emulate `consume` based on `get`. But I think this would simply duplicate what `get` does for no actual benefit, and so is not needed.

Among the existing implementation in the library, I'm quite sure some of them are incompatible with this method.